### PR TITLE
🏗️(backends) Unify data backends

### PR DESCRIPTION
**Pull request by SergioSim**
## Purpose

Storage and Database backends have similar interfaces and usage. 

## Proposal

Unify the interfaces of database and storage backends into a single DataBackend interface.

- [X] add Data and LRS Backend interfaces
- [X] add FileSystemDataBackend
- [ ] add LDPDataBackend
- [ ] add SwiftDataBackend
- [ ] add ESDataBackend
- [ ] add MongoDataBackend
- [ ] replace database and storage backends with data backends

### CODE REVIEW

File: src/ralph/backends/data/fs.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
+"""FileSystem data backend for Ralph"""
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply src/ralph/backends/data/fs.py:1]: Thanks)

File: src/ralph/backends/data/fs.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+            logger.info(msg, self.default_directory)
+            self.default_directory.mkdir(parents=True)
+
+        logger.debug("Default directory: %s", self.default_directory)
+
+    def status(self) -> DataBackendStatus:
+        """Checks whether the default directory has appropriate write permissions."""
+
+        if os.access(self.default_directory, os.W_OK):
+            return DataBackendStatus.OK
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply src/ralph/backends/data/fs.py:78]: Well spotted) Thanks) Indeed, that's more readable)

File: src/ralph/backends/data/fs.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+        return DataBackendStatus.ERROR
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Generator[None, Union[str, dict], None]:
+        """Lists files and directories contained in the target directory.
+
+        Args:
+            target (str or None): The directory path where to list the files.
+                If target is `None`, the `default_directory_path` is used instead.
```
[the remark this answers is not in the document]
Comment (SergioSim) [reply src/ralph/backends/data/fs.py:89]: Well spotted) Thanks)

**Comment by SergioSim** [comment]
Closing this PR in favor of the rebased #228 PR

**Comment by jmaupetit** [comment]
Closed in favor of #228

**Pull request closed without merging**